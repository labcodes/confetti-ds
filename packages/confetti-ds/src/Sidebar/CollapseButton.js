import React from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";

export default class CollapseButton extends React.Component {
  static propTypes = {
    /** Callback action to be executed when the CollapseButton is clicked. */
    onClick: PropTypes.func.isRequired,
  };

  render() {
    const { onClick } = this.props;

    return (
      <button
        type="button"
        onClick={onClick}
        className="lab-narrow-sidebar__collapse"
      >
        <Icon
          type="menu-collapse"
          className="lab-narrow-sidebar__collapse-icon"
        />
      </button>
    );
  }
}
