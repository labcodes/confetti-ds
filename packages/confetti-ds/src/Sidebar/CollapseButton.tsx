import React from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";

/**
 *
 * @param onClick Callback action to be executed when the CollapseButton is clicked.
 * @returns {JSX.Element}
 * @constructor
 *
 */
export default function CollapseButton({ onClick }) {
  return (
    <button
      type="button"
      onClick={onClick}
      className="lab-narrow-sidebar__collapse"
    >
      <Icon
        type="menu-collapse"
        className="lab-narrow-sidebar__collapse-icon"
      />
    </button>
  );
}

CollapseButton.propTypes = {
  /** Callback action to be executed when the CollapseButton is clicked. */
  onClick: PropTypes.func.isRequired,
};
