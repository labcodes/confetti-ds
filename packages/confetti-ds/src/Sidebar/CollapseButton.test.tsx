import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import CollapseButton from "./CollapseButton";

describe("CollapseButton", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<CollapseButton onClick={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowButton = shallow(<CollapseButton onClick={mockOnClick} />);
    shallowButton.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
