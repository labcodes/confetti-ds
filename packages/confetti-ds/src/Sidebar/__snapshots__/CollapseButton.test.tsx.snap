// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseButton renders with base props 1`] = `
<button
  className="lab-narrow-sidebar__collapse"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-icon lab-icon--menu-collapse lab-narrow-sidebar__collapse-icon"
  >
    <svg
      aria-hidden={true}
      className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
      fill="currentColor"
      height={20}
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3.5 11h5a.5.5 0 0 1 .5.41v5.09a.5.5 0 0 1-1 .09V12.7l-5.15 5.15a.5.5 0 0 1-.76-.63l.06-.07L7.29 12H3.5a.5.5 0 0 1-.5-.41v-.09a.5.5 0 0 1 .41-.5H8.5h-5Zm14.35-8.85c.18.17.2.44.06.63l-.06.07L12.71 8h3.79a.5.5 0 0 1 .5.41v.09a.5.5 0 0 1-.41.5h-5.18l-.1-.04-.08-.04-.05-.04-.03-.03-.04-.04-.04-.06-.03-.06-.02-.06V8.6a.5.5 0 0 1-.02-.1v.07V3.5a.5.5 0 0 1 1-.09V7.3l5.15-5.15c.2-.2.5-.2.7 0Z"
        fill="currentColor"
      />
    </svg>
  </span>
</button>
`;
